tez-dag/src/main/java/org/apache/tez/dag/app/dag/impl/VertexImpl.java

```diff
@@ -2843,7 +2855,7 @@ private void assignVertexManager() throws TezException {
     if (recoveryData != null && recoveryData.shouldSkipInit()
```

Comment:
this whole condition reached a complexity where we should move it into a separate method like:
```
canSkipVertexInitOnRecovery(recoveryData)
```

Reply:
I can do it once we agree on the course of TEZ-4569. Do you prefer to move the whole condition in this `if` branch?

Reply:
Done. I'm ready to update more if you have a different thought or additional requests.
https://github.com/apache/tez/pull/361/commits/c8c18fcb49de1d0cadd368ed8db73750f9d154f7